python/ray/tune/tuner.py

```diff
@@ -270,7 +270,7 @@ def fit(self) -> ResultGrid:
         if not self._is_ray_client:
             try:
                 return self._local_tuner.fit()
-            except Exception as e:
+            except TuneError as e:
```

Comment:
Does this affect any other cases? I.e. are there other paths that are not raising a `TuneError` but are expected to be intercepted here?

Reply:
Not sure. I'm hoping tests will cover this if there are any. @xwjiang2010 do you know?

Reply:
Not sure either but looks like some tests are failing. Maybe start from there?